File: 0189-rotate-array/0189-rotate-array.py
class Solution:
    def rotate(self, nums: list[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n=len(nums)
        k=k%n
        nums[:] = nums[::-1]
        nums[:k] = nums[:k][::-1]
        nums[k:] = nums[k:][::-1]

File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int n=nums.size();
        k=k%n;
        vector<int>temp(n);
        int i=n-k;
        int j=0;
        for(i=n-k;i<n;i++){
            temp[j]=nums[i];
            j++;
        }
        for(i = 0; i < n-k; i++) {
            temp[j] = nums[i];
            j++;
        }
        for(i = 0; i < n; i++) {
            nums[i] = temp[i];
        }
    }
};